Managers/NetworkServer.cpp

```diff
@@ -1680,45 +1647,65 @@ namespace RTE {
 
 			// Process reset votes
 			int votesNeeded = 0;
-			int votes = 0;
+			int endActivityVotes = 0;
+			int restartVotes = 0;
+
 			for (short player = 0; player < c_MaxClients; player++) {
 				if (IsPlayerConnected(player)) {
 					votesNeeded++;
-					if (m_ResetActivityVotes[player]) { votes++; }
+					if (m_EndActivityVotes[player]) { endActivityVotes++; }
+					if (m_RestartActivityVotes[player]) { restartVotes++; }
 				}
 			}
-			if (votes > 0) {
-				char buf[128];
 
-				std::snprintf(buf, sizeof(buf), "Voting to end activity %d of %d", votes, votesNeeded);
+			std::string displayMsg = "";
+
+			if (endActivityVotes > 0) {
+				displayMsg = "Voting to end activity: " + std::to_string(endActivityVotes) + " of " + std::to_string(votesNeeded);
+			}
+			if (restartVotes > 0) {
+				if (!displayMsg.empty()) { displayMsg += "\n"; }
+				displayMsg += "Voting to restart activity: " + std::to_string(restartVotes) + " of " + std::to_string(votesNeeded);
+			}
+			
+			if (votesNeeded > 0 && (endActivityVotes > 0 || restartVotes > 0)) {
 				for (short i = 0; i < c_MaxClients; i++) {
-					g_FrameMan.SetScreenText(buf, i, 0, -1, false);
+					g_FrameMan.SetScreenText(displayMsg, g_ActivityMan.GetActivity()->ScreenOfPlayer(i), 0, 500);
+					g_ActivityMan.GetActivity()->ResetMessageTimer(i);
 				}
 
-			}
-			if (votes == votesNeeded && votesNeeded > 0) {
-				for (short player = 0; player < c_MaxClients; player++) {
-					m_ResetActivityVotes[player] = false;
+				if (endActivityVotes >= votesNeeded) {
+					// Only reset gameplay activities, and not server lobby
```

Comment:
I wonder if we should even show the reset message if we're in the lobby activity, since it doesn't do anything there.